firebase-vscode/src/core/config.ts

```diff
@@ -104,112 +112,207 @@ function notifyFirebaseConfig(broker: ExtensionBrokerImpl) {
+
+  const configWatcher = await _createWatcher(firebaseJsonPattern);
+  // Track the URI of any firebase.json in the project.
+  if (configWatcher) {
+    context.subscriptions.push(configWatcher);
+
+    // We don't listen to changes here, as we'll only watch the selected config.
+    configWatcher.onDidCreate((addedUri) => {
+      console.log("here", "addedUri", allFirebaseConfigsUris.value);
+      allFirebaseConfigsUris.value = [
+        ...allFirebaseConfigsUris.value,
+        addedUri,
+      ];
+    });
+    configWatcher.onDidDelete((deletedUri) => {
+      console.log("here", "removedUri", allFirebaseConfigsUris.value);
+      allFirebaseConfigsUris.value = allFirebaseConfigsUris.value.filter(
+        (uri) => uri.fsPath !== deletedUri.fsPath,
+      );
+    });
+  }
 
-  const firebaseConfigRemoveListener = onChange(firebaseConfig, () =>
-    notifyFirebaseConfig(broker),
-  );
+  context.subscriptions.push({
+    dispose: onChange(firebaseConfig, () => notifyFirebaseConfig(broker)),
+  });
 
-  const showToastOnError = effect(() => {
-    const config = firebaseConfig.value;
-    if (config instanceof ResultError) {
-      vscode.window.showErrorMessage(
-        `Error reading firebase.json:\n${config.error}`,
-      );
-    }
+  // When no config is selected, or the selected config is deleted, select the first one.
+  context.subscriptions.push({
+    dispose: effect(() => {
+      const configUri = selectedFirebaseConfigUri.value;
+      // We watch all config URIs before selecting one, so that when deleting the selected
+      // config, the effect runs again and selects a new one.
+      const allConfigUris = allFirebaseConfigsUris.value;
+      console.log("here", "selecting firebase config", allConfigUris);
+      if (configUri && fs.existsSync(configUri.fsPath)) {
+        console.log("here", "selected config still valid");
+        return;
+      }
+
+      if (allConfigUris[0] !== selectedFirebaseConfigUri.value) {
+        selectedFirebaseConfigUri.value = allConfigUris[0];
+      }
+    }),
   });
 
-  const configWatcher = _createWatcher("firebase.json");
-  configWatcher?.onDidChange(
-    () => (firebaseConfig.value = _readFirebaseConfig()),
-  );
-  configWatcher?.onDidCreate(
-    () => (firebaseConfig.value = _readFirebaseConfig()),
-  );
-  configWatcher?.onDidDelete(() => (firebaseConfig.value = undefined));
+  let disposable: Disposable | undefined;
+  context.subscriptions.push({ dispose: () => disposable?.dispose() });
+  context.subscriptions.push({
+    dispose: effect(() => {
+      disposable?.dispose();
+      disposable = undefined;
+      firebaseRC.value = undefined;
+
+      const configUri = selectedFirebaseConfigUri.value;
+      console.log("here", "configUri", configUri);
+      if (!configUri) {
+        return;
+      }
+
+      disposable = configWatcher?.onDidChange((uri) => {
+        if (uri.fsPath !== configUri.fsPath) {
```

Comment:
should this be 

if (a === b) { firebaseConfig.value = _readFirebaseConfig(configUri);}

Reply:
Good catch